core/vm/interpreter.go

```diff
@@ -259,22 +259,16 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 
 		// execute the operation
 		res, err = operation.execute(&pc, in, callContext)
-		// if the operation clears the return data (e.g. it has returning data)
-		// set the last return to the result of the operation.
-		if operation.returns {
-			in.returnData = res
-		}
 
-		switch {
-		case err != nil:
-			return nil, err
-		case operation.reverts:
-			return res, ErrExecutionReverted
-		case operation.halts:
-			return res, nil
-		case !operation.jumps:
-			pc++
+		if err != nil {
+			break
 		}
+		pc++
 	}
-	return nil, nil
+
+	if err == errStopToken {
```

Comment:
more go-idiomatic
```suggestion
	if errors.Is(err, errStopToken) {
```
But maybe your variant is more performant?

Reply:
The `errors.Is()` is not inlined (https://godbolt.org/z/n81qdKTzn), but this does not matter because this check is outside of the loop.

Reply:
My slight preference is to leave `==` because of other usages in EVM, e.g. `suberr == ErrCodeStoreOutOfGas`.

Reply:
There is no need to use `errors.Is` here. The error is purely internal, and this check is performance-critical. 
`errors.Is` is appropriate when dealing with unknown and potentially nested errors.